package/Runtime/Scripts/Internal/NimbusAdUnit.cs

```diff
@@ -138,8 +138,8 @@ public sealed class NimbusAdUnit {
 
 		internal event DestroyAdDelegate OnDestroyIOSAd;
 
-		private void OnOnDestroyIOSAd() {
-			OnDestroyIOSAd?.Invoke();
+		private void DestroyIOSAd() {
```

Comment:
This has to be called within the Public destroy method fyi, IOS can't have a different public API for destroying ads.